b-lab-org/ember-cli-impact-core
# [Wellness] Remove eslint-plugin-ember overrides

#### Overview
This is follow up from https://github.com/b-lab-org/ember-cli-impact-core/pull/96 - it removes the remaining eslint rules that were overridden to keep the scope smaller.

#### How to test
Test in the context of the impact-app and manage-app. An alternative could be to `yarn link` if you don't want to pull all branches.

Things to look out for:
- question interactions - choosing an option should toggle the color of the action button and the inverse
- any `permission` model (assessment, question) should still work

#### Notes and lessons learned
- Removing observers was challenging, but led to cleaner code and fixed a few bugs indirectly.
- I specifically disabled the [named-functions](https://github.com/ember-cli/eslint-plugin-ember/blob/HEAD/docs/rules/named-functions-in-promises.md) rule for tests. This threw a lot of exceptions for `return wait().then(() => { });` code.
- Testing DOM interactions with things like chosen, the datepicker is **incredibly** difficult. I brought in [ember-native-dom-helpers](https://github.com/cibernox/ember-native-dom-helpers) to help - but didn't have much luck. I left the dependency because I think down the road it would be worthwhile to explore what it would take to remove jquery.
